awx/main/access.py

```diff
@@ -2243,6 +2254,7 @@ class NotificationAccess(BaseAccess):
 
     def filtered_queryset(self):
         return self.model.objects.filter(
+            Q(notification_template__organization__in=Organization.objects.filter(notification_admin_role__members=self.user)) |
             Q(notification_template__organization__in=self.user.admin_of_organizations) |
```

Comment:
Again, same type of comment as I left inside of NotificationTemplateAccess class, try replacing the 2 OR conditions by 1 condition

```python
Q(notification_template__organization__in=Organization.accessible_objects(self.user, 'notification_admin_role')
```